modin/core/dataframe/pandas/exchange/dataframe_protocol/column.py

```diff
@@ -0,0 +1,614 @@
+    TBD: there's also the "chunk" concept here, which is implicit in Arrow as
+         multiple buffers per array (= column here). Semantically it may make
+         sense to have both: chunks were meant for example for lazy evaluation
+         of data which doesn't fit in memory, while multiple buffers per column
+         could also come from doing a selection operation on a single
+         contiguous buffer.
+         Given these concepts, one would expect chunks to be all of the same
+         size (say a 10,000 row dataframe could have 10 chunks of 1,000 rows),
+         while multiple buffers could have data-dependent lengths. Not an issue
+         in pandas if one column is backed by a single NumPy array, but in
+         Arrow it seems possible.
+         Are multiple chunks *and* multiple buffers per column necessary for
+         the purposes of this interchange protocol, or must producers either
+         reuse the chunk concept for this or copy the data?
+
+    Parameters
+    ----------
+    column : PandasDataframe
+        A ``PandasDataframe`` object.
+    allow_copy : bool, default: True
+        A keyword that defines whether or not the library is allowed
+        to make a copy of the data. For example, copying data would be necessary
+        if a library supports strided buffers, given that this protocol
+        specifies contiguous buffers. Currently, if the flag is set to ``False``
+        and a copy is needed, a ``RuntimeError`` will be raised.
+    offset : int, default: 0
+        The offset of the first element.
+
+    Notes
+    -----
+    This Column object can only be produced by ``__dataframe__``,
+    so doesn't need its own version or ``__column__`` protocol.
+    """
+
+    def __init__(
+        self, column: PandasDataframe, allow_copy: bool = True, offset: int = 0
+    ) -> None:
+        if not isinstance(column, PandasDataframe):
+            raise NotImplementedError(
+                "Columns of type {} not handled " "yet".format(type(column))
+            )
+
+        # Store the column as a private attribute
+        self._col = column
+        self._allow_copy = allow_copy
+        self._offset = offset
+
+    @property
+    def size(self) -> int:
+        """
+        Size of the column, in elements.
+
+        Corresponds to `DataFrame.num_rows()` if column is a single chunk;
+        equal to size of this current chunk otherwise.
+
+        Returns
+        -------
+        int
+            Size of the column, in elements.
+        """
+        return len(self._col.index)
+
+    @property
+    def offset(self) -> int:
+        """
+        Get the offset of first element.
+
+        May be > 0 if using chunks; for example for a column
+        with N chunks of equal size M (only the last chunk may be shorter),
+        ``offset = n * M``, ``n = 0 .. N-1``.
+
+        Returns
+        -------
+        int
+            The offset of first element.
+        """
+        return self._offset
```

Comment:
This should always be 0. Going from the discussion given in data-apis/dataframe-api#67, the `offset` value indicates the offset in the actual buffers that are being returned from `get_buffers()`. 

For instance, if all the chunks are backed by a single buffer then `.get_buffers()` return the same for each chunk, when the `offset` indicates the number of bits (`offset * dtype.bit_width`) to start interpreting the data from in this particular chunk. Since in ModinOnPandas implementation chunks are independent buffers there is no such thing as an offset.

Reply:
Given the discussion

>> Yeah, so that's indeed ambiguous in the spec: is the offset only informative for where the chunked Column fits in the >> full Column, or does it determine how to interpret the Buffer?

> The latter.

now we return 0.